### PR TITLE
Features/14.2

### PR DESCRIPTION
**Pull request by AndreiBrehuescu**
Created the search functionality. Now when searching the API from Open Route Service uses the autocomplete to find a match for the partial string.

### CODE REVIEW

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx [hunk 1]
```diff
@@ -24,36 +24,75 @@ interface Props {
   searchMarker: LatLngExpression[];
   setsearchMarker: (searchMarker: LatLngExpression[]) => void;
 }
+
+function getUserDataWithPromise(props: string) {
+  const searchString =
+    'https://api.openrouteservice.org/geocode/autocomplete?api_key=' +
```
Comment (AndreiSold) [inline event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx]: You should extract this url in a separate constanst file

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx [hunk 2]
```diff
@@ -24,36 +24,75 @@ interface Props {
...
+
 const SearchBar = (props: Props) => {
   const classesSearch = useStylesSearchBar();
   const [flag, setFlag] = useState(true);
   const [suggestions, setSuggestions] = useState<LocationType[]>([]);
+  const [locations, setLocations] = useState<LocationType[]>([]);
   const { t } = useTranslation();
 
   useEffect(() => {
+    const l: LocationType[] = [];
```
Comment (AndreiSold) [inline event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx]: You should be more specific with the naming of variables, maybe `locationTypes` here

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBarSuggestions.tsx [hunk 3]
```diff
@@ -1,34 +1,32 @@
...
 const RenderSuggestions = (props: Props) => {
   const classesSearch = useStylesSearchBar();
 
-  if (props.suggestions.length === 0) {
-    return null;
-  } else {
-    if (props.suggestions.length > 4) {
-      const firstSuggestions = props.suggestions.slice(0, 4);
+  useEffect(() => {
+    if (props.suggestions.length > 5) {
```
Comment (AndreiSold) [inline event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBarSuggestions.tsx]: This 5 should also be extracted in that constants file (`MAX_SUGGESTIONS_SHOWED` or something like that) so you can easily change it without searching for it within the code

**Review by AndreiSold — APPROVED** [review]

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx (hunk 1 above)
Comment (AndreiBrehuescu) [reply event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx]: Fixed

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx (hunk 2 above)
Comment (AndreiBrehuescu) [reply event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx]: Changed to foundLocations

File: event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBarSuggestions.tsx (hunk 3 above)
Comment (AndreiBrehuescu) [reply event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBarSuggestions.tsx]: Fixed

**Pull request merged**